src/passes/shadowstack.ts

```diff
@@ -477,7 +478,11 @@ export class ShadowStackPass extends Pass {
     let moduleRef = this.module.ref;
     _BinaryenRemoveFunction(moduleRef, name);
     let cArr = allocPtrArray(vars);
-    _BinaryenAddFunction(moduleRef, name, params, results, cArr, vars.length, body);
+    let newFuncRef = _BinaryenAddFunction(moduleRef, name, params, results, cArr, vars.length, body);
+    if (this.options.sourceMap || this.options.debugInfo) {
+      let func = Function.searchFunctionByRef(this.compiler, newFuncRef);
```

Comment:
The shadow stack pass has a reference to the `Program` here via the compiler. With this information, I think `Function.searchFunctionByRef` could be simplified to `program.searchFunctionByRef`?